test/skimNtuple2016_HHbtag.cpp

```diff
@@ -2668,598 +2581,430 @@ int main (int argc, char** argv)
-      vector<float> idAndIsoSF_leg2_deep_vsEle_down    (2, idAndIsoSF_leg2_deep_vsEle);    // in bins of eta: barrel, endcap
-      vector<float> idAndIsoSF_leg2_deep_vsMu_down     (5, idAndIsoSF_leg2_deep_vsMu);     // in bins of eta, edges at 0, 0.4, 0.8, 1.2, 1.7, infty
-
-      for (int bin = 0; bin < (int) isthisPt_IDbin_second.size(); bin++)
-      {
-	if (isthisPt_IDbin_second[bin])
-	{
-	  idAndIsoSF_leg2_deep_vsJet_pt_up[bin]   = Deep_antiJet_medium_pt ->getSFvsPT(tau2pt, tau2Genmatch,   "Up");
-	  idAndIsoSF_leg2_deep_vsJet_pt_down[bin] = Deep_antiJet_medium_pt ->getSFvsPT(tau2pt, tau2Genmatch, "Down");
+	float leg1pt  = tlv_firstLepton.Pt();
+	float leg1eta = TMath::Abs(tlv_firstLepton.Eta());
+	float leg2pt  = tlv_secondLepton.Pt();
+	float leg2eta = TMath::Abs(tlv_secondLepton.Eta());
+
+	// the first tau only makes sense in the TauTau channel
+	float tau1DM  = theSmallTree.m_dau1_decayMode;
+	float tau2DM  = theSmallTree.m_dau2_decayMode;
+	float tau1Genmatch = theBigTree.genmatch->at(firstDaughterIndex);
+	float tau2Genmatch = theBigTree.genmatch->at(secondDaughterIndex);
+
+	float idSF_leg1				  = 1.f;
+	float idSF_leg1_deep_vsJet_dm = 1.f;
+	float idSF_leg1_deep_vsJet_pt = 1.f;
+	float idSF_leg1_deep_vsJet_2d = 1.f;
+	float idSF_leg1_deep_vsEle    = 1.f;
+	float idSF_leg1_deep_vsMu     = 1.f;
+
+	float idSF_leg2				  = 1.f;
+	float idSF_leg2_deep_vsJet_dm = 1.f;
+	float idSF_leg2_deep_vsJet_pt = 1.f;
+	float idSF_leg2_deep_vsJet_2d = 1.f;
+	float idSF_leg2_deep_vsEle	  = 1.f;
+	float idSF_leg2_deep_vsMu	  = 1.f;
+
+	bool isFakeJet1 = true;
+    bool isFakeJet2 = true;
+	if (tau1Genmatch<6) {
+	  isFakeJet1 = false;
 	}
-      }
-
-      for (int bin = 0; bin < (int) isthisEta_IDbin_second.size(); bin++)
-      {
-	if (isthisEta_IDbin_second[bin])
-	{
-	  idAndIsoSF_leg2_deep_vsMu_up[bin]   = Deep_antiMu_tight ->getSFvsEta(tau2eta, tau2Genmatch,   "Up");
-	  idAndIsoSF_leg2_deep_vsMu_down[bin] = Deep_antiMu_tight ->getSFvsEta(tau2eta, tau2Genmatch, "Down");
+	if (tau2Genmatch<6) {
+	  isFakeJet2 = false;
 	}
-      }
 
-      for (int bin = 0; bin < (int) isthisSDet_IDbin_second.size(); bin++)
-      {
-	if (isthisSDet_IDbin_second[bin])
-	{
-	  idAndIsoSF_leg2_deep_vsEle_up[bin]   = Deep_antiEle_vloose ->getSFvsEta(tau2eta, tau2Genmatch,   "Up");
-	  idAndIsoSF_leg2_deep_vsEle_down[bin] = Deep_antiEle_vloose ->getSFvsEta(tau2eta, tau2Genmatch, "Down");
-	}
-      }
-
-      if (tau2Genmatch==1 || tau2Genmatch==2 || tau2Genmatch==3 || tau2Genmatch==4 || tau2Genmatch==5)
-      {
-	isFakeJet2 = false;
-      }
+	// only TauTau has a tau as the first leg
+	if (isMC and pType==2)
+	  {
+		idSF_leg1_deep_vsJet_dm = Deep_antiJet_medium_dm->getSFvsDM(leg1pt, tau1DM, tau1Genmatch);
+		idSF_leg1_deep_vsJet_pt = Deep_antiJet_medium_pt->getSFvsPT(leg1pt, tau1Genmatch);
+		idSF_leg2_deep_vsJet_2d = Deep_antiJet_2d->getSFvsDMandPT(leg2pt, tau2DM, tau2Genmatch);
```

Comment:
I suspect this should be leg1 and not leg2, probably also in the other skimmers

Reply:
Nice catch, I had forgotten to update this in 2016 and 2017.